$OpenBSD: patch-Alc_mixer_sse_c,v 1.1 2016/05/24 08:07:58 dcoppa Exp $

Avoid an underflow in the SSE mixers if BufferSize is less than 4
(upstream git commit 564a1822595e287cf808ea63a03d96719e3cc7b4)

--- Alc/mixer_sse.c.orig	Mon May 23 20:59:18 2016
+++ Alc/mixer_sse.c	Mon May 23 21:00:32 2016
@@ -155,7 +155,7 @@ void MixDirect_SSE(const DirectParams *params, const A
             ClickRemoval[c] -= data[0]*DrySend;
 
         gain = _mm_set1_ps(DrySend);
-        for(pos = 0;pos < BufferSize-3;pos += 4)
+        for(pos = 0;BufferSize-pos > 3;pos += 4)
         {
             const __m128 val4 = _mm_load_ps(&data[pos]);
             __m128 dry4 = _mm_load_ps(&DryBuffer[c][OutPos+pos]);
@@ -189,7 +189,7 @@ void MixSend_SSE(const SendParams *params, const ALflo
         WetClickRemoval[0] -= data[0] * WetGain;
 
     gain = _mm_set1_ps(WetGain);
-    for(pos = 0;pos < BufferSize-3;pos+=4)
+    for(pos = 0;BufferSize-pos > 3;pos += 4)
     {
         const __m128 val4 = _mm_load_ps(&data[pos]);
         __m128 wet4 = _mm_load_ps(&WetBuffer[0][OutPos+pos]);
